test/hold_effect_clear_amulet.c

```diff
@@ -0,0 +1,57 @@
+#include "global.h"
+#include "test_battle.h"
+
+ASSUMPTIONS
+{
+    ASSUME(gItems[ITEM_CLEAR_AMULET].holdEffect == HOLD_EFFECT_CLEAR_AMULET);
+}
+
+SINGLE_BATTLE_TEST("Clear Amulet prevents Intimidate")
+{
+    s16 turnOneHit;
+    s16 turnTwoHit;
+
+    GIVEN {
+        PLAYER(SPECIES_EKANS) { Ability(ABILITY_SHED_SKIN); };
+        PLAYER(SPECIES_EKANS) { Ability(ABILITY_INTIMIDATE); };
+        OPPONENT(SPECIES_WOBBUFFET) { Item(ITEM_CLEAR_AMULET); };
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_TACKLE); }
+        TURN { SWITCH(player, 1); MOVE(opponent, MOVE_TACKLE); }
+    } SCENE {
+        HP_BAR(player, captureDamage: &turnOneHit);
+        ABILITY_POPUP(player, ABILITY_INTIMIDATE);
+        NONE_OF { ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, player); }
+        MESSAGE("Foe Wobbuffet's Attack was not lowered!");
+        HP_BAR(player, captureDamage: &turnTwoHit);
+    } THEN {
+        EXPECT_EQ(turnOneHit, turnTwoHit);
+    }
+}
+
+SINGLE_BATTLE_TEST("Clear Amulet prevents stat reducing effects")
+{
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_GROWL].effect == EFFECT_ATTACK_DOWN);
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET) { Item(ITEM_CLEAR_AMULET); };
+    } WHEN {
+        TURN { MOVE(player, MOVE_GROWL); }
+    } SCENE {
+        NONE_OF { ANIMATION(ANIM_TYPE_GENERAL, B_ANIM_STATS_CHANGE, opponent); }
+        MESSAGE("Foe Wobbuffet's Attack was not lowered!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Clear Amulet prevents secondary effects that reduce stats")
+{
```

Comment:
To make sure every stat change is handled, let's use `PARAMETRIZE` to add:
- Aurora Beam (Attack)
- Snarl (Sp. Attack)
- Psychic (Sp. Defense)
- Bubble Beam (Speed)
- Mud Slap (Accuracy)

Unfortunately, there's no move for evasion that does this.